ToolBox/AlgoSeekOptionsConverter/Program.cs

```diff
@@ -0,0 +1,407 @@
+            this.OptionType = detail[0].Trim();
+            this.Strike = ConvertStringToLong(detail[2]);
+            this.Expiration = detail[4].Replace("-", "").Trim();
+        }
+
+        private long ConvertStringToLong(string str)
+        {
+            return (long)(10000 * decimal.Parse(str.Trim(), NumberStyles.Any, new CultureInfo("en-US")));
+        }
+
+        public string SetFilename(string resolution)
+        {
+            return this.IsInvalid ? "Invalid tick" :
+                string.Format(@"{0}\{1}\{2}\{3}\{4:yyyyMMdd}_{1}_{0}_{2}_{3}_{5}_{6}.csv", resolution,
+                this.Underlying, this.OptionType, this.Expiration, this.Time, this.Strike,
+                this.TickType == "TRADE" ? "trade" : "quote").ToLower();
+        }
+
+        public string CsvFileOutput()
+        {
+            return this.Time.TimeOfDay.TotalMilliseconds.ToString("F0") + "," + this.Price + "," + this.Quantity;
+        }
+
+        public override string ToString()
+        {
+            return this.IsInvalid ? "Invalid tick" :
+                (this.Underlying + " " + this.OptionType + " " + this.Expiration + " " + this.Strike + " " +
+                this.Time.ToString(@"yyyyMMdd hhmmss\.fff") + " " + this.Price + " " + this.Quantity + " " + this.TickType).ToLower();
+        }
+    }
+    /// <summary>
+    /// QuoteBar class
+    /// </summary>
+    class QuoteBar
+    {
+        public DateTime Time { get; set; }
+        public string Underlying { get; set; }
+        public string OptionType { get; set; }
+        public long Strike { get; set; }
+        public string Expiration { get; set; }
+
+        public long BidOpen { get; set; }
+        public long BidHigh { get; set; }
+        public long BidLow { get; set; }
+        public long BidClose { get; set; }
+        public long AskOpen { get; set; }
+        public long AskHigh { get; set; }
+        public long AskLow { get; set; }
+        public long AskClose { get; set; }
+        public long AvgBidSize { get; set; }
+        public long AvgAskSize { get; set; }
+
+        public string CsvFileOutput()
+        {
+            var bidstr = this.AvgBidSize == 0 ? ",,,," : string.Format("{0},{1},{2},{3},{4}",
+                this.BidOpen, this.BidHigh, this.BidLow, this.BidClose, this.AvgBidSize);
+
+            var askstr = this.AvgAskSize == 0 ? ",,,," : string.Format("{0},{1},{2},{3},{4}",
+                this.AskOpen, this.AskHigh, this.AskLow, this.AskClose, this.AvgAskSize);
+
+            return this.Time.TimeOfDay.TotalMilliseconds.ToString("F0") + "," + bidstr + "," + askstr;
+        }
+
+        public override string ToString()
+        {
+            return this.Underlying + " " + this.OptionType + " " + this.Expiration + " " + this.Strike + " " + this.Time.ToString("yyyyMMdd") + " " + this.CsvFileOutput();
+        }
+    }
+    /// <summary>
+    /// TradeBar class
+    /// </summary>
+    class TradeBar
```

Comment:
Same here (can't use this name), I think we should figure out the new name/design for the option-tradebar/quotebar types and update this writer to create those types directly. Will discuss in the options-issue,